tripal/src/TripalDBX/TripalDbxSchema.php

```diff
@@ -690,6 +690,38 @@ public function primaryKeyExists(
     return $constraint_exists;
   }
 
+  /**
+   * If a FK exists between two tables, return the left and right columns.
+   *
+   * @param string $left_table
+   *   The name of the table the foreign key resides in. E.g. 'feature' for
+   *   the feature.type_id => cvterm.cvterm_id foreign key.
+   * @param string $right_table
+   */
+  public function getForeignKeyDef(string $left_table, string $right_table) {
+    $left_def = $this->getTableDef($left_table, ['format' => 'Drupal']);
+    if (!array_key_exists($right_table, $left_def['foreign keys'])) {
+      return NULL;
+    }
+    return $left_def['foreign keys'][$right_table];
+  }
+
+  /**
+   * If a FK exists between two tables, return the left and right columns.
+   *
+   * @param string $left_table
+   *   The name of the table the foreign key resides in. E.g. 'feature' for
+   *   the feature.type_id => cvterm.cvterm_id foreign key.
+   * @param string $right_table
+   */
+  public function foreignKeyExists(string $left_table, string $right_table) {
+    $left_def = $this->getForeignKeyDef($left_table, $right_table);
+    if ($left_def) {
+      return TRUE;
+    }
+    return FALSE;
+  }
+
```

Comment:
```suggestion
  /**
   * If a FK exists between two tables, return the left and right columns.
   *
   * @param string $left_table
   *   The name of the table the foreign key resides in. E.g. 'feature' for
   *   the feature.type_id => cvterm.cvterm_id foreign key.
   * @param string $right_table
   *   The name of the table the foreign key refers to. For the example
   *   above it would be cvterm.
   * @return array
   *    The foreign key definition
   */
  public function getForeignKeyDef(string $left_table, string $right_table) {
    $left_def = $this->getTableDef($left_table, ['format' => 'Drupal']);
    if (!array_key_exists($right_table, $left_def['foreign keys'])) {
      return NULL;
    }
    return $left_def['foreign keys'][$right_table];
  }

  /**
   * Indicates whether a FK exists between two tables.
   *
   * @param string $left_table
   *   The name of the table the foreign key resides in. E.g. 'feature' for
   *   the feature.type_id => cvterm.cvterm_id foreign key.
   * @param string $right_table
   *   The name of the table the foreign key refers to. For the example
   *   above it would be cvterm.
   * @return bool
   *   TRUE if a foreign key exists, FALSE if not.
   */
  public function foreignKeyExists(string $left_table, string $right_table) {
    $left_def = $this->getForeignKeyDef($left_table, $right_table);
    if ($left_def) {
      return TRUE;
    }
    return FALSE;
  }

```
Let me know if this suggestion is correct!!! Does the first function return an array or something else?